File: src/main/java/medium/TopFrequentElements.java
package medium;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class TopFrequentElements {
  public int[] solve(int[] nums, int k) {
    int[] result = new int[k];
    HashMap<Integer, Integer> freq = new HashMap<>();
    for (int i : nums) {
      freq.put(i, freq.getOrDefault(i, 0) + 1);
    }
    List<Map.Entry<Integer, Integer>> sorted =
        freq.entrySet().stream().sorted((a, b) -> b.getValue() - a.getValue()).toList();
    System.err.println(sorted);

    for (int i = 0; i < result.length; i++) {
      result[i] = sorted.get(i).getKey();
    }
    return result;
  }
}

File: src/medium/top_frequent_elements.rs
use std::collections::HashMap;

pub fn top_frequent_elements(nums: Vec<i32>, k: i32) -> Vec<i32> {
    let mut result: Vec<i32> = vec![0; k as usize];
    let mut freq: HashMap<i32, i32> = HashMap::new();
    for n in nums {
        freq.entry(n)
            .and_modify(|c| {
                *c += 1;
            })
            .or_insert(1);
    }
    let mut sorted: Vec<(&i32, &i32)> = freq.iter().collect();
    sorted.sort_by(|a, b| b.1.cmp(a.1));
    for i in 0..result.len() {
        result[i] = *sorted[i].0;
    }
    result
}
